universal-login-sdk/lib/core/services/WalletService.ts

```diff
@@ -59,6 +62,21 @@ export class WalletService {
     return futureWallet;
   }
 
+  async deployFutureWallet(ensName: string, gasPrice: string, gasToken: string) {
+    if (this.state.kind !== 'Future') {
+      throw new Error('Invalid state: expected future wallet');
+    }
+    const {deploy, contractAddress, privateKey} = this.state.wallet;
+
+    const applicationWallet = {contractAddress, name: ensName, privateKey};
+    this.stateProperty.set({kind: 'Deploying', wallet: applicationWallet});
```

Comment:
Why not keep it as FutureWallet?

Reply:
Future wallet doesn't have a name yet, when deploying name is already determined, therefore ApplicationWallet

Reply:
Ens name is determined when we deployed a wallet. If deploying throws an error, we should in the future provide the ability to deploy the wallet with another ensName